pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/BasePatch.java

```diff
@@ -279,18 +280,15 @@ protected List<NameValue> setWorkflowRelatedFields(@NonNull Patch patch) {
                                     patch.getNameValues(), patch.getDdpParticipantId()), "sent"));
                 }
             }
-        } else if (patch.getNameValue().getName().equals("oD.unableObtainTissue") && !(boolean) patch.getNameValue().getValue()) {
-            boolean hasReceivedDate = new OncHistoryDetailDaoImpl().hasReceivedDate(getOncHistoryDetailId(patch));
-
-            if (hasReceivedDate) {
-                nameValues.add(setAdditionalValue("oD.request",
-                        new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "received"));
-            } else {
-                nameValues.add(setAdditionalValue("oD.request",
-                        new Patch(patch.getId(), PARTICIPANT_ID, patch.getParentId(), patch.getUser(), patch.getNameValue(),
-                                patch.getNameValues(), patch.getDdpParticipantId()), "sent"));
-            }
+        } else if (patch.getNameValue().getName().equals("oD.unableObtainTissue")) {
+            String status = (boolean) patch.getNameValue().getValue()
+                    ? OncHistoryDetail.UNABLE_OBTAIN_TISSUE
```

Comment:
Improved, but this construct is hard to understand.
Perhaps an old school if statement would be easier to read?

Reply:
added a comment which should help understanding it.